src/main/java/com/checkmarx/flow/service/ReposManagerService.java

```diff
@@ -0,0 +1,55 @@
+package com.checkmarx.flow.service;
+
+import com.checkmarx.flow.config.CxIntegrationsProperties;
+import com.checkmarx.flow.config.external.CxGoDynamicConfig;
+import com.checkmarx.flow.exception.ReposManagerException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import javax.annotation.PostConstruct;
+
+@Service
+@Slf4j
+public class ReposManagerService {
+
+    private final RestTemplate restTemplate;
+    private final CxIntegrationsProperties cxIntegrationsProperties;
+
+    private String cxGoConfigUrlPattern;
+
+    public ReposManagerService(@Qualifier("flowRestTemplate") RestTemplate restTemplate,
+                               CxIntegrationsProperties cxIntegrationsProperties) {
+        this.restTemplate = restTemplate;
+        this.cxIntegrationsProperties = cxIntegrationsProperties;
+    }
+
+    @PostConstruct
+    private void compositeUrlPaths() {
+        cxGoConfigUrlPattern = cxIntegrationsProperties.getUrl() + "/%s/orgs/%s/tenantConfig";
```

Comment:
Why is this logic extracted into PostConstruct?

Reply:
I found it useful, similarly to Asaf repos manager's logic